airflow/api_connexion/endpoints/dag_endpoint.py

```diff
@@ -66,6 +66,11 @@ def get_dag_details(*, dag_id: str) -> APIResponse:
     dag: DAG = get_airflow_app().dag_bag.get_dag(dag_id)
     if not dag:
         raise NotFound("DAG not found", detail=f"The DAG with dag_id: {dag_id} was not found")
+    dag_model: DagModel = DagModel.get_dagmodel(dag_id=dag_id)
+    for key, value in dag_model.__dict__.items():
+        if not key.startswith("_") and not hasattr(dag, key):
+            setattr(dag, key, value)
```

Comment:
Not sure if modifying the DAG object in-place is the right thing to do, I think the DAG bag reuses the objects in different calls. The opposite is likely better, fetching a DagModel object first and set attributes from a DAG.

Reply:
Would it make more sense to deepcopy the dag object?

Reply:
Yes but why not use the DagModel object instead? That would also better match other DAG endpoints, which all use the db model instead.